pkg/frontend/querymiddleware/spin_off_subqueries_queryable.go

```diff
@@ -80,6 +145,11 @@ func (q *spinOffSubqueriesQuerier) Select(ctx context.Context, _ bool, hints *st
 			return storage.ErrSeriesSet(errors.New("missing required labels for downstream query"))
 		}
 
+		set := q.loadFromCacheOrWait(ctx, query)
+		if set != nil {
+			return set
+		}
+
```

Comment:
If the query fails after `loadFromCacheOrWait` returns `nil` (both in this `case` branch or the subquery spinoff branch below), then don't we need to do something to unblock any other `loadFromCacheOrWait` calls that might be waiting for this result? Otherwise they won't return until the query times out and the context is cancelled.